web-server/v0.4/utils/datastore.js

```diff
@@ -0,0 +1,6 @@
+export default { 
+    run_indices: ".run.*",
```

Comment:
Won't this mean we will be searching all indices that match "`.run.*`"?  Can you we use this value to construct a set of indices like (pseudo-code below):
```
function construct_es_indices_for_query(prefix, index_pattern, time_range) {
    if pattern_in_days(index_pattern) {
        dates = get_list_of_dates_in_days(time_range, index_pattern)
    } else if pattern_in_months(index_pattern) {
        dates = get_list_of_dates_in_months(time_range, index_pattern)
    } else {
        // Assume pattern does not include a time stamp
        return prefix + "." + index_pattern;
    }
    indices = "";
    for date in dates {
        indices += prefix + "." + index_pattern.format(date) + ","; 
    }
    return indices;
}
```

Reply:
Unfortunately, the old architecture does not take user input for the time range for run indices. For the purpose of correctly documenting this PR, I left the wildcard index for run data as this is reflective of the functionality allowed by this version. However, this design has been improved in the new architecture where I've used Redux to reference the global state in order to lookup the time range a user has selected before issuing a query to elasticsearch. Let me know your thoughts - I'd be happy to add this functionality to the old architecture as I know using the wildcard is not great for performance. 

Below is the utility function I've implemented to parse the user's selected time range in the new architecture: 

```
function parseMonths(indexPrefix, startMonth, endMonth) {
  let months = '';

  if (!startMonth.isBefore(endMonth)) {
    months = months.concat(indexPrefix + startMonth.format('YYYY-MM') + ',');
  }
  while (startMonth.isBefore(endMonth) && startMonth.isBefore(moment().endOf('month'))) {
    months = months.concat(indexPrefix + startMonth.format('YYYY-MM') + ',');
    startMonth.add(1, 'month');
  }

  return months;
}
```

Reply:
@gurbirkalsi, that seems reasonable, we can leave this as is for this patch, and wait for the next version to solve the wildcard index problems.